aws/resource_aws_cognito_user_pool.go

```diff
@@ -0,0 +1,902 @@
+				MinItems: 0,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"external_id": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"sns_caller_arn": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validateArn,
+						},
+					},
+				},
+			},
+
+			"sms_verification_message": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateCognitoUserPoolSmsVerificationMessage,
+			},
+
+			"tags": tagsSchema(),
+
+			"username_attributes": {
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+					ValidateFunc: validation.StringInSlice([]string{
+						cognitoidentityprovider.UsernameAttributeTypeEmail,
+						cognitoidentityprovider.UsernameAttributeTypePhoneNumber,
+					}, false),
+				},
+				ConflictsWith: []string{"alias_attributes"},
+			},
+
+			"verification_message_template": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"default_email_option": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Default:      cognitoidentityprovider.DefaultEmailOptionTypeConfirmWithCode,
+							ValidateFunc: validateCognitoUserPoolTemplateDefaultEmailOption,
+						},
+						"email_message": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Computed:     true,
+							ValidateFunc: validateCognitoUserPoolTemplateEmailMessage,
+						},
+						"email_message_by_link": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Computed:     true,
+							ValidateFunc: validateCognitoUserPoolTemplateEmailMessageByLink,
+						},
+						"email_subject": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Computed:     true,
+							ValidateFunc: validateCognitoUserPoolTemplateEmailSubject,
+						},
+						"email_subject_by_link": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Computed:     true,
+							ValidateFunc: validateCognitoUserPoolTemplateEmailSubjectByLink,
+						},
+						"sms_message": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Computed:     true,
+							ValidateFunc: validateCognitoUserPoolTemplateSmsMessage,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceAwsCognitoUserPoolCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).cognitoidpconn
+
+	params := &cognitoidentityprovider.CreateUserPoolInput{
+		PoolName: aws.String(d.Get("name").(string)),
+	}
+
+	if v, ok := d.GetOk("admin_create_user_config"); ok {
+		configs := v.([]interface{})
+		config, ok := configs[0].(map[string]interface{})
+
+		if !ok {
+			return errors.New("admin_create_user_config is <nil>")
```

Comment:
Nitpick, but I'd just ignore empty `admin_create_user_config` block, basically I'd just modify the condition below to something like 

```go
if ok && config != nil {
```
It's what I'd probably expect more as a user.